File: src/main/java/net/tianben/advancedcombat/item/sword/CreativeSwordItem.java
package net.tianben.advancedcombat.item.sword;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.ItemStack;
import net.minecraft.item.SwordItem;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.Text;
import net.minecraft.world.GameMode;
import net.minecraft.world.World;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class CreativeSwordItem extends SwordItem {
    public CreativeSwordItem() {
        super(CreativeMaterial.CREATIVE, 3, -2.4f, new FabricItemSettings());
    }

    @Override
    public boolean postHit(ItemStack stack, LivingEntity target, LivingEntity attacker) {
        if (attacker instanceof PlayerEntity player) {
            if (checkGamemode(player)) {
                if (!target.getWorld().isClient) {
                    target.discard();
                }
            }
        }
        return super.postHit(stack, target, attacker);
    }

    private boolean checkGamemode(PlayerEntity player) {
        if (player instanceof ServerPlayerEntity serverPlayer) {
            return serverPlayer.interactionManager.getGameMode() == GameMode.CREATIVE;
        } else if (player.getWorld().isClient && player instanceof PlayerEntity) {
            MinecraftClient client = MinecraftClient.getInstance();
            return client.getNetworkHandler().getPlayerListEntry(player.getUuid()) != null &&
                    client.getNetworkHandler().getPlayerListEntry(player.getUuid()).getGameMode() == GameMode.CREATIVE;
        }
        return false;
    }

    public void appendTooltip(ItemStack stack, @Nullable World world, List<Text> tooltip, TooltipContext context) {
        super.appendTooltip(stack, world, tooltip, context);
        tooltip.add(Text.of("§aActive"));
    }
}

File: src/main/kotlin/net/tianben/advancedcombat/item/sword/CreativeSwordItem.kt
package net.tianben.advancedcombat.item.sword

import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.block.BlockState
import net.minecraft.client.MinecraftClient
import net.minecraft.client.item.TooltipContext
import net.minecraft.entity.Entity
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack
import net.minecraft.item.SwordItem
import net.minecraft.item.ToolMaterial
import net.minecraft.text.Text
import net.minecraft.util.Formatting
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

class CreativeSwordItem(
    material: ToolMaterial,
    attackDamage: Int,
    attackSpeed: Float,
    settings: Settings
) : SwordItem(material, attackDamage, attackSpeed, settings) {

    override fun postHit(stack: ItemStack, target: LivingEntity, attacker: LivingEntity): Boolean {
        if (attacker is PlayerEntity && attacker.isCreative) {
            target.kill()
            target.remove(Entity.RemovalReason.KILLED)
            return true
        }
        return super.postHit(stack, target, attacker)
    }

    override fun inventoryTick(stack: ItemStack, world: World, entity: Entity, slot: Int, selected: Boolean) {
        if (entity !is PlayerEntity || !entity.isCreative) {
            super.inventoryTick(stack, world, entity, slot, selected)
        } else {
            if (stack.damage > 0) stack.damage = 0
        }
    }

    override fun postMine(stack: ItemStack, world: World, state: BlockState, pos: BlockPos, miner: LivingEntity): Boolean {
        return if (miner is PlayerEntity && miner.isCreative) {
            true
        } else {
            super.postMine(stack, world, state, pos, miner)
        }
    }

    override fun canMine(state: BlockState, world: World, pos: BlockPos, miner: PlayerEntity): Boolean {
        return miner.isCreative || super.canMine(state, world, pos, miner)
    }

    @Environment(EnvType.CLIENT)
    override fun hasGlint(stack: ItemStack): Boolean {
        val player = MinecraftClient.getInstance().player
        return player != null && player.isCreative
    }

    @Environment(EnvType.CLIENT)
    override fun appendTooltip(stack: ItemStack, world: World?, tooltip: MutableList<Text>, context: TooltipContext) {
        val player = MinecraftClient.getInstance().player
        if (player != null && player.isCreative) {
            tooltip.add(Text.translatable("tooltip.advancedcombat.active").formatted(Formatting.GREEN))
        } else {
            tooltip.add(Text.translatable("tooltip.advancedcombat.inactive").formatted(Formatting.RED))
        }
    }
}